tests/v2/validation/provisioning/rke2/etcd_node_delete_test.go

```diff
@@ -0,0 +1,162 @@
+	client                 *rancher.Client
+	ns                     string
+	rke2kubernetesVersions []string
+	cnis                   []string
+	providers              []string
+	nodesAndRoles          []machinepools.NodeRoles
+}
+
+func (e *EtcdNodeDeleteTestSuite) TearDownSuite() {
+	e.session.Cleanup()
+}
+
+func (e *EtcdNodeDeleteTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	e.session = testSession
+
+	e.ns = defaultNS
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	e.rke2kubernetesVersions = clustersConfig.RKE2KubernetesVersions
+
+	e.cnis = clustersConfig.CNIs
+	e.providers = clustersConfig.Providers
+	e.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(e.T(), err)
+
+	e.client = client
+}
+
+func (e *EtcdNodeDeleteTestSuite) isEtcdNodeRemoved(query url.Values, clusterName string, etcdNodeToDelete v1.SteveAPIObject) (int, error) {
+	noOfEtcdNodesAfterDeletion := 0
+
+	err := kwait.Poll(5*time.Second, 15*time.Minute, func() (done bool, err error) {
+		machines, err := e.client.Steve.SteveType(machineSteveResourceType).List(query)
+		if err != nil {
+			return false, err
+		}
+
+		noOfEtcdNodesAfterDeletion = 0
+		for _, machine := range machines.Data {
+			if machine.Labels[etcdLabel] == "true" && machine.Labels[clusterLabel] == clusterName {
+				if machine.Name == etcdNodeToDelete.Name {
+					return false, nil
+				}
+				noOfEtcdNodesAfterDeletion++
+			}
+		}
+		logrus.Info("new etcd node : ")
+		for _, machine := range machines.Data {
+			if machine.Labels[etcdLabel] == "true" && machine.Labels[clusterLabel] == clusterName {
+				logrus.Info(machine.Name)
+			}
+		}
+		return true, nil
+	})
+	return noOfEtcdNodesAfterDeletion, err
+}
+
+func (e *EtcdNodeDeleteTestSuite) TestEtcdDeleteNode() {
+	require.GreaterOrEqual(e.T(), len(e.providers), 1)
+	require.GreaterOrEqual(e.T(), len(e.cnis), 1)
+	require.GreaterOrEqual(e.T(), len(e.rke2kubernetesVersions), 1)
+
+	subSession := e.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := e.client.WithSession(subSession)
+	require.NoError(e.T(), err)
+
+	for _, provider := range e.providers {
+		for _, k8sVersion := range e.rke2kubernetesVersions {
+			for _, cni := range e.cnis {
+				provider := CreateProvider(provider)
+				logrus.Info("provisioning ", k8sVersion, " cluster using ", cni, " CNI..")
+				clusterResp := TestProvisioningRKE2Cluster(e.T(), client, provider, e.nodesAndRoles, k8sVersion, cni)
+
+				query, err := url.ParseQuery("")
+				require.NoError(e.T(), err)
+
+				query["fieldSelector"] = []string{"metadata.namespace=" + e.ns}
+
+				machines, err := e.client.Steve.SteveType(machineSteveResourceType).List(query)
+				require.NoError(e.T(), err)
+
+				noOfEtcdNodesBeforeDeletion := 0
+				etcdNodeToDelete := v1.SteveAPIObject{}
+
+				logrus.Info("etcd nodes before deletion : ")
```

Comment:
Same comment as before, I do not have a strong preference either way. But I think this logging can just be removed as you're already logging the etcd node to be removed.

Reply:
Removed it.